### ISSUE TITLE
Issues with login to VW servers

### ISSUE THREAD

**Issue by mtjoernelund** [tillsteinbach/WeConnect-mqtt#77]
I just noticed the new version with fix for a problem logging in to VW servers. I just pulled the latest docker image but still log the following error.

`2022-01-31T11:03:25+0000:ERROR:weconnect_mqtt_base:Could not connect to VW-Server: HTTPSConnectionPool(host='login.apps.emea.vwapps.io', port=443): Max retries exceeded with url: /authorize?nonce=IEhCTAKTwIexnyxD&redirect_uri=weconnect://authenticated (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f1fb990a100>: Failed to establish a new connection: [Errno -3] Try again')), will retry in 10 seconds `

Should I just wait awhile for the login attempts to reset and try again, or is this potentially another issue?

**Comment by tillsteinbach** [comment]
This feels like there is no internet connection.

**Comment by mtjoernelund** [comment]
I think you are right. Network in Docker seem to have locked up. Restarting docker seem to have done the trick.

**Comment by mtjoernelund** [comment]
Reopening this issue. Something weird seems to be going on, and I am not sure where to look for possible problems. I thought that it was my docker installation that was acting up, but after running for some time WeConnect-mqtt seem to lock up. There is no output to the log and according to docker the container is still running, but no messages are coming through to my mutt broker. I tried restarting just the container and still no messages. Other containers are running just fine. So I tried recreating the container using host network mode instead of using reconnects own docker network and now it is running. Will see for how long. But I have a suspicion that it may actually be the docker network created that locks up rather than the entire container.

**Comment by tillsteinbach** [comment]
you can also increase the output with `-v` or  `-vv` or  `-vvv`

**Comment by tillsteinbach** [comment]
Here is an example: https://github.com/tillsteinbach/WeConnect-mqtt/blob/main/docker/docker-compose.yml

**Comment by MichaelSt71** [comment]
I still have a problem with my docker image. Was there a change at VW server or login mechanism? About 2 or 3 month ago everything works fine. But then one day to another this message came up:
"2022-05-14T08:57:41+0000:CRITICAL:weconnect_mqtt_base:There was a problem when communicating with WeConnect.  If this problem persists please open a bug report: No credentials form found"

**Comment by tillsteinbach** [comment]
Are you on the latest version? VW constantly changes things on their servers. This makes it necessary to update from time to time.

**Comment by MichaelSt71** [comment]
Yes I am. I've installed latest docker Image this morning.

**Comment by MichaelSt71** [comment]
Additionally information: we have an ID.3 from beginning this year.
I already tried to get information about the API from weconnect support, but they told me nothing and were very tight-lipped.

**Comment by tillsteinbach** [comment]
Ok, better do not bother the WeConnect support. This is an inofficial API that we are using. 

can you increase the loglevel for me by adding -vvv and give me the output?

**Comment by MichaelSt71** [comment]
Ok, here's the output

2022-05-15T09:17:52+0000:INFO:weconnect_mqtt_base:WeConnect-mqtt 0.26.1 (using WeConnect-python 0.32.1)
2022-05-15T09:17:52+0000:INFO:weconnect_mqtt_base:Connect to WeConnect
2022-05-15T09:17:52+0000:DEBUG:connectionpool:Starting new HTTPS connection (1): login.apps.emea.vwapps.io:443
2022-05-15T09:17:52+0000:DEBUG:connectionpool:https://login.apps.emea.vwapps.io:443 "GET /authorize?nonce=dQKI6YiLZWmrpDG0&redirect_uri=weconnect://authenticated HTTP/1.1" 303 0
2022-05-15T09:17:52+0000:DEBUG:connectionpool:Starting new HTTPS connection (1): identity.vwgroup.io:443
2022-05-15T09:17:52+0000:DEBUG:connectionpool:https://identity.vwgroup.io:443 "GET /oidc/v1/authorize?client_id=a24fba63-34b3-4d43-b181-942111e6bda8%40apps_vw-dilab_com&scope=openid+profile+badge+cars+dealers+vin&response_type=code+id_token+token&redirect_uri=weconnect%3A%2F%2Fauthenticated&state=97d17c37b3b9bfbce5ca52c39a9f693887cb6e66faeb76b4d4b4e7cc77b3bf25&nonce=dQKI6YiLZWmrpDG0 HTTP/1.1" 302 0
2022-05-15T09:17:52+0000:DEBUG:connectionpool:https://identity.vwgroup.io:443 "GET /signin-service/v1/signin/a24fba63-34b3-4d43-b181-942111e6bda8@apps_vw-dilab_com?relayState=d8bde1ba0ce6798d3365cf5ef1b5318278c88988 HTTP/1.1" 200 None
2022-05-15T09:17:52+0000:DEBUG:connectionpool:https://identity.vwgroup.io:443 "POST /signin-service/v1/a24fba63-34b3-4d43-b181-942111e6bda8@apps_vw-dilab_com/login/identifier HTTP/1.1" 303 0
2022-05-15T09:17:52+0000:DEBUG:connectionpool:https://identity.vwgroup.io:443 "GET /signin-service/v1/a24fba63-34b3-4d43-b181-942111e6bda8@apps_vw-dilab_com/login/authenticate?relayState=d8bde1ba0ce6798d3365cf5ef1b5318278c88988 HTTP/1.1" 200 None
2022-05-15T09:17:52+0000:CRITICAL:weconnect_mqtt_base:There was a problem when communicating with WeConnect.  If this problem persists please open a bug report: No credentials form found

**Comment by tillsteinbach** [comment]
The version you are using is outdated. Looks like something went wrong updating the container.

**Comment by MichaelSt71** [comment]
Yes, your right. But I've installed the latest image: "tillsteinbach/weconnect-mqtt:latest". Docker container on a QNAP NAS.
Now I've switched to python command line. Everything works fine. Thanks!
One last question: why are so many topics under error code
code=4001
message=Internal error
info=Internal error, please try again later. If the problem persists, please contact our support.

**Comment by tillsteinbach** [comment]
Regarding the docker: maybe you just recreated the container without pulling the image again?
The 4001 is because of the firmware of your ID3. Those sections will become available with the 3.0 update of your car.

**Comment by MichaelSt71** [comment]
Great!
